test/bioprinttest.c

```diff
@@ -11,8 +11,12 @@
 
 #include <stdio.h>
 #include <string.h>
-#include <openssl/opensslconf.h>
-#include OPENSSL_UNISTD
+#include <openssl/e_os2.h>
+#ifdef OPENSSL_SYS_WINDOWS
+# include <winsock.h>
```

Comment:
Besides, MSVC defines write in io.h, and preferred way to refer to it is _write. And I don't see no write in winsock.h. WinSock interface doesn't have no write, it has send/recv, you can also use WriteFile on socket, but you can't use write with socket. Because it accepts _fileno(fp), which has *nothing* to do with kind of descriptors used by send/recv or WriteFile.